src/App.js

```diff
@@ -43,18 +45,22 @@ function App() {
     console.log(cartProducts)
   };
 
-  const onAddToFavorite = (obj) => {
-    setFavorites(prev => {
-      // Check if the item already exists in favorites
-      if (!prev.some(item => item.id === obj.id)) {
-        // If it doesn't exist, add it to the favorites array
-        return [...prev, obj];
-      }
-      // If it already exists, return the unchanged favorites array
-      return prev;
-    });
 
-  };
+  const onAddToFavorite = async (obj) => {
```

Comment:
looks like function name does not reflect its logic
It's not only adding, it's adding or deleting. Better to use verb "toggle"